src/browser/tab/mod.rs

```diff
@@ -242,19 +240,13 @@ impl<'a> Tab {
         };
 
         if node_ids.is_empty() {
-            return Err(NoElementFound {
-                selector: selector.to_string(),
-            }
-            .into());
-        }
-
-        let mut elements = vec![];
-
-        for node_id in &node_ids {
-            elements.push(Element::new(&self, *node_id, selector)?)
+            return Err(NoElementFound {}.into());
         }
 
-        Ok(elements)
+        node_ids
+            .into_iter()
+            .map(|node_id| Element::new(&self, node_id))
+            .collect()
```

Comment:
Note that one can `.collect()` into a `Result<T<U>, E>` where `T: FromIterator<Item=Result<U, E>>`, a highly useful shortcut.